ext/bz2: Check int params of bzcompress() are correct

Girgias opened the pull request:
Also add a TODO to check the length of the source strings

nielsdos reviewed (approved):
This is a subtle BC break though, so I'd recommend documenting this in UPGRADING.
Also, shouldn't this be tested?

nielsdos reviewed (changes requested):
In fact yeah, this needs tests and also existing test changes...

nielsdos commented:
See for example https://github.com/php/php-src/blob/dcbba04d38ff11b7a9bb34efc2d380d9c55ee608/ext/bz2/tests/005.phpt#L15

Girgias commented:
> This is a subtle BC break though, so I'd recommend documenting this in UPGRADING. Also, shouldn't this be tested?

I wanted to make this a draft as I was having dinner.
But I should have addressed the issues.
